client/jetpack-connect/authorize.js

```diff
@@ -881,29 +891,33 @@ export class JetpackAuthorize extends Component {
 		// 'jetpack/magic-link-signup' feature flag) are created with a username based on the user's
 		// email address. For this reason, we want to display both the username and the email address
 		// so users can start making the connection between the two immediately. Otherwise, users might
-		// not recognize their username since they didn't created it.
+		// not recognize their username since they didn't create it.
+
+		const connected = authorizeSuccess || this.props.isAlreadyOnSitesList;
+		const connectionStatus = connected ? 'Connected' : 'Connecting';
+
+		// We make an exception for Woo onboardings, since in these cases the creation of a Jetpack account
+		// is an intermediate step and the user will be redirected to the WooCommerce onboarding flow.
+		// Seeing this new username/email address can cause confusion because they have already set up
+		// a Woo account under their own email address.
+		if ( ( isWooOnboarding || isWooPaymentsOnboarding ) && isJetpackMagicLinkSignUpFlow ) {
```

Comment:
I just took a deeper look at the code. The `from=woocommerce-onboarding` parameter is added to the URL when users use WC <=7.8. so it's not needed anymore. We should go ahead to deprecate this parameter. I'll create an issue for this.

I think we can just use `wooDna.isWooDnaFlow()` to determine if the user is from WooPayments, Woo Shipping, or other woo flows.

Reply:
> Could we use wooDna.isWooDnaFlow() here? It determines if the user is from WooPayments, Woo Shipping, or other woo jetpack connection flows.

Yes, you're right - I checked and we can use Woo DNA to determine if it came from WooPayments (also confirmed this on a JN site) and this should also work for Woo Shipping and any other plugins which implement the WooDNA flow in future. Updated in 9609d7391c